### DISCUSSION TITLE
sysmonforlinux

### DISCUSSION THREAD

**Discussion by hodan112** [Security-Onion-Solutions/securityonion#14028] [Unsupported Versions]
Hello everyone,
I’m currently working with Security Onion 2.4.110, and I’m trying to integrate SysmonForLinux logs for hunting. However, I’m having some difficulty with parsing these logs properly for analysis in Security Onion.

SysmonForLinux Log Parsing:

I’ve installed SysmonForLinux and configured it to send logs to elasticsearch. But I’m struggling to properly parse and visualize these logs for effective threat hunting.

Question: What are the best practices or steps to properly parse SysmonForLinux logs in elasticsearch? Are there specific configurations or processors I need to set up in order to effectively analyze these logs?

Any guidance, tips, or documentation links would be greatly appreciated!

Thank you in advance for your help!

Best regards,
Hodan Dirieh

**Comment by dougburks** [comment]
Have you tried the Elastic Agent for Linux? It provides much of the same hunting capability as sysmon logs.

https://docs.securityonion.net/en/2.4/downloads.html
https://docs.securityonion.net/en/2.4/elastic-agent.html

**Reply by hodan112** [reply]
Hello ,
I have integrated SysmonForLinux logs into Elasticsearch, and I’m already using Elastic Agent for log ingestion. The logs are being ingested correctly in XML format, and I would like to know if it is possible to process and apply playbooks or analytical rules to these logs for enhanced threat hunting.

**Reply by defensivedepth** [reply]
Greetings @hodan112 - to be clear, you are seeing the logs correctly ingested now?

**Reply by hodan112** [reply]
Yes, the logs have been ingested correctly, and we can see them in Kibana, but they are in XML format.
<img width="937" alt="sysmon" src="https://github.com/user-attachments/assets/6310e787-11e8-4c10-a237-a975bd44da97" />

**Reply by defensivedepth** [reply]
They are being ingested, but not correctly. Please post the elastic agent integration configuration that is being used.

**Reply by hodan112** [reply]
I created this ingestion pipeline: https://gist.github.com/mttaggart/efc870e02eb603943e0dae8ebd54d3dc 

and I added a configuration to the elastic-agent.yml file. Here’s the configuration used:
<img width="482" alt="image" src="https://github.com/user-attachments/assets/fac371bc-c675-4340-a944-73c9ec009963" />

Let me know if adjustments are needed.

**Reply by hodan112** [reply]
still waiting

**Comment by hodan112** [comment]
Hello @defensivedepth ,
Thank you for your assistance, I have successfully mapped the Sysmon for Linux logs.
<img width="356" alt="image" src="https://github.com/user-attachments/assets/a5359280-a47d-44bb-9b31-a29304377623" />

However, I would appreciate your feedback to confirm if the method I used is correct or if there are other alternatives.
Process:

1.Logs from Linux systems are collected through the "system" integration, which parses the system syslog logs using the "logs-system.syslog" pipeline.
2.I applied a custom pipeline to parse only the XML field, which is renamed to "message." This pipeline is specifically designed to process this XML field.
<img width="337" alt="image" src="https://github.com/user-attachments/assets/797e4bcb-daca-4124-896c-45d18360c8df" />

I would greatly appreciate it if you could confirm whether this approach is correct or if there are other options to consider.
